sdk-base/src/main/java/com/mapbox/maps/plugin/viewport/experimental/transition/ViewportTransitionOptions.kt

```diff
@@ -0,0 +1,78 @@
+package com.mapbox.maps.plugin.viewport.experimental.transition
+
+import com.mapbox.maps.plugin.viewport.DEFAULT_FRAME_TRANSITION_MAX_DURATION_MS
+
+/**
+ * Options that impact the transition animation.
+ *
+ * The styling of the animation is determined by the [ViewportStateTransition]
+ * and [ViewportTransition] implementations, but options here provide higher-level
+ * constraints that those implementations need to obey.
+ *
+ * @param maxDurationMs maximum duration of the generated transitions set,
+ * including delays between animators and their respective durations.
+ */
+class ViewportTransitionOptions private constructor(
```

Comment:
It only contains one property, do we really need this class?

Reply:
we can debate it, previously this class contains two camera options, one for following state and one for overview state. We can probably remove this class and use camera options instead.

Reply:
btw why not data class? similar to `ViewportData`

Reply:
@LukasPaczos could you suggest why it's constructed this way in the beginning? I think navigation team use builder pattern a lot and not data classes.

Reply:
This class was aimed to provide more options in the future, if needed.

It's not a data class because you cannot add a new property to a data class without breaking compatibility, especially for Java consumers. Example in https://jakewharton.com/public-api-challenges-in-kotlin/#copy-functions.

Reply:
I have updated the PR to remove usage of data classes in the options for future additions.